Modules/Sources/PointOfSale/ViewHelpers/PointOfSaleDashboardViewHelper.swift

```diff
@@ -1,12 +1,34 @@
 import Foundation
 import SwiftUI
+import enum Yosemite.POSCatalogSyncState
 
 struct PointOfSaleDashboardViewHelper {
     static func determineViewState(
         eligibilityState: POSEligibilityState?,
         itemsContainerState: ItemsContainerState,
-        horizontalSizeClass: UserInterfaceSizeClass?
+        horizontalSizeClass: UserInterfaceSizeClass?,
+        catalogSyncState: POSCatalogSyncState?
     ) -> PointOfSaleDashboardView.ViewState {
+
+        /// Check first to show syncing state as soon as possible
+        if let syncState = catalogSyncState {
+            switch syncState {
+            case .syncStarted(_, true):
+                return .catalogSyncing
+            case .syncNeverDone:
+                return .catalogSyncing
+            case .syncStarted(_, false):
+                // Non-initial sync, continue to other checks
+                break
+            case .syncCompleted:
+                // Continue to other checks
+                break
+            case .syncFailed:
+                // TODO: WOOMOB-1565
+                return .error(PointOfSaleErrorState.errorOnLoadingOrders())
+            }
+        }
```

Comment:
Perhaps we'd avoid the need for this (and some of the async stream stuff) if the GRDBItemsController handled this and set the container state to `.loading(.catalogSyncing)`? I know it's a big change, and I could be missing some fundamental reason why it won't work or would be a bad way to go...

Reply:
> Do you see how it can be avoided if we implemented the logic within the controller?

No, thinking about it more we'd still need something like the stream, which should definitely still be in the coordinator, so that we can show the current sync if there's one already ongoing.

We'd be reading that stream in the GRDBItemsController, and setting the container state there too. I didn't really know that we used this helper, to be honest... it's probably fine in either place.